sos/plugins/gnocchi.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (C) 2016 Red Hat, Inc., Sachin Patil <sacpatil@redhat.com>
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+from sos.plugins import Plugin, RedHatPlugin
+
+
+class GnocchiPlugin(Plugin, RedHatPlugin):
+    """Gnocchi - Metric as a service"""
+    plugin_name = "gnocchi"
+    profiles = ('system', 'services')
+    requires_root = False
+
+    files = (
+        '/etc/gnocchi/gnocchi.conf',
+        '/etc/gnocchi/gnocchi-config-generator.conf',
+        '/etc/gnocchi/policy.json',
+        '/etc/gnocchi/api-paste.ini',
+    )
+
+    packages = ('gnocchi',)
+
+    def setup(self):
+        self.add_cmd_output([
+            "gnocchi --version",
+            "gnocchi status",
+            "gnocchi capabilities list",
+            "gnocchi archive-policy list",
+            "gnocchi resource list",
+            "gnocchi resource-type list"
+        ])
```

Comment:
I see that the `gnocchi` command accepts authentication information via the `--os-username`, `--os-password`, and related environment variables (`OS_USER` etc.).

Do we expect that all installations will have this disabled? If not, and the `gnocchi` shell command prompts for them interactively then `sos` will take a timeout (default 300s, 5m) on each `gnocchi` invocation.

If we do expect to find authenticated installations then the plugin needs to be able to handle that: e.g. by passing in values in environment variables (*not* using plugin options - these are exposed in `/proc` to unprivileged users).